### PR TITLE
Support ARB with Flint 3

### PR DESCRIPTION
**Pull request by isuruf**
Arb is now a part of flint

Fixes https://github.com/symengine/symengine/issues/1987

### CODE REVIEW

File: cmake/FindFLINT.cmake [hunk 1]
```diff
@@ -4,11 +4,21 @@ libfind_include(flint/flint.h flint)
...
 set(FLINT_LIBRARIES ${FLINT_LIBRARY})
-set(FLINT_INCLUDE_DIRS ${FLINT_INCLUDE_DIR})
+# Workaround for https://github.com/fredrik-johansson/arb/issues/24
+set(FLINT_INCLUDE_DIRS ${FLINT_INCLUDE_DIR} ${FLINT_INCLUDE_DIR}/flint)
+
 set(FLINT_TARGETS flint)
 
+file(READ "${FLINT_INCLUDE_DIR}/flint/flint.h" FLINT_H)
+string(REGEX MATCH "#define FLINT_VERSION \"([0-9.]*)\"" _ ${FLINT_H})
+set(FLINT_VERSION ${CMAKE_MATCH_1})
```
Comment (isuruf) [inline cmake/FindFLINT.cmake:14]: @antonio-rojas, can you print the value of `${FLINT_VERSION}` here?

**Comment by isuruf** [comment]
Looking at the Arch Linux build I see the following
```
/* flint version number */

#define __FLINT_VERSION 3
#define __FLINT_VERSION_MINOR 0
#define __FLINT_VERSION_PATCHLEVEL 1
#define FLINT_VERSION ""
#define __FLINT_RELEASE_NUM(a,b,c) ((a)*10000 + (b)*100 + (c))
#define __FLINT_RELEASE __FLINT_RELEASE_NUM(__FLINT_VERSION, __FLINT_VERSION_MINOR, __FLINT_VERSION_PATCHLEVEL)
```

I'm not sure how `FLINT_VERSION` became empty.

**Comment by isuruf** [comment]
@antonio-rojas, you need https://github.com/flintlib/flint/commit/88bcb2eee75c15e2f99c4634cbd4467c04fdb204 backported.

**Comment by antonio-rojas** [comment]
Thanks. Still doesn't build with flint 3 as it doesn't account for API changes, see https://gitlab.archlinux.org/archlinux/packaging/packages/symengine/-/blob/main/flint-3.patch?ref_type=heads#L29-61

**Comment by isuruf** [comment]
I think you also need https://github.com/symengine/symengine/commit/1ef98559d75b4fa5b87dad2daff0d12e7857ac8d

**Comment by antonio-rojas** [comment]
Missed that, sorry. Tried with master now to make sure there's no other commit missing, but it looks like there's still a missing `fmpq.h` include in `ntheory.cpp`
```
/build/symengine/src/symengine/symengine/ntheory.cpp: In function ‘SymEngine::RCP<const SymEngine::Number> SymEngine::bernoulli(long unsigned int)’:
/build/symengine/src/symengine/symengine/ntheory.cpp:499:5: error: ‘fmpq_init’ was not declared in this scope; did you mean ‘fmpz_init’?
  499 |     fmpq_init(res);
      |     ^~~~~~~~~
      |     fmpz_init
In file included from /build/symengine/src/symengine/symengine/visitor.h:15,
                 from /build/symengine/src/symengine/symengine/test_visitors.h:4,
                 from /build/symengine/src/symengine/symengine/pow.cpp:5:
/build/symengine/src/symengine/symengine/series_generic.h: In constructor ‘SymEngine::UnivariateSeries::UnivariateSeries(const SymEngine::UExprDict&, std::string, unsigned int)’:
/build/symengine/src/symengine/symengine/series_generic.h:27:31: warning: redundant move in initialization [-Wredundant-move]
   27 |         : SeriesBase(std::move(sp), varname, degree)
      |                      ~~~~~~~~~^~~~
/build/symengine/src/symengine/symengine/series_generic.h:27:31: note: remove ‘std::move’ call
/build/symengine/src/symengine/symengine/ntheory.cpp:503:5: error: ‘fmpq_get_mpq’ was not declared in this scope; did you mean ‘fmpz_get_mpz’?
  503 |     fmpq_get_mpq(a, res);
      |     ^~~~~~~~~~~~
      |     fmpz_get_mpz
In file included from /build/symengine/src/symengine/symengine/series_generic.h:11:
/build/symengine/src/symengine/symengine/series.h: At global scope:
/build/symengine/src/symengine/symengine/series.h:128:23: warning: ‘SymEngine::RCP<const SymEngine::Number> SymEngine::SeriesBase<Poly, Coeff, Series>::pow(const SymEngine::Number&) const [with Poly = SymEngine::fmpq_poly_wrapper; Coeff = SymEngine::fmpq_wrapper; Series = SymEngine::URatPSeriesFlint]’ was hidden [-Woverloaded-virtual=]
  128 |     RCP<const Number> pow(const Number &other) const override
      |                       ^~~
In file included from /build/symengine/src/symengine/symengine/visitor.h:17:
/build/symengine/src/symengine/symengine/series_flint.h:41:18: note:   by ‘static SymEngine::fqp_t SymEngine::URatPSeriesFlint::pow(const SymEngine::fqp_t&, int, unsigned int)’
   41 |     static fqp_t pow(const fqp_t &s, int n, unsigned prec);
      |                  ^~~
/build/symengine/src/symengine/symengine/ntheory.cpp:505:5: error: ‘fmpq_clear’ was not declared in this scope; did you mean ‘fmpz_clear’?
  505 |     fmpq_clear(res);
      |     ^~~~~~~~~~
      |     fmpz_clear
```

**Comment by isuruf** [comment]
No worries. Thanks for checking.

**Comment by antonio-rojas** [comment]
All good now.

**Review by certik — APPROVED** [review]

**Pull request merged**